item2/Dictionary.java

```diff
@@ -0,0 +1,53 @@
+import java.util.*;
+
+public class Dictionary {
+  private Map<String, Object> dict = new HashMap<String, Object>();
+
+  private Dictionary(Builder builder) {
+    dict = builder.dict;
+  }
+
+  @Override
+  public String toString() {
+    String str = "{\n";
```

Comment:
Do you think string builder is more efficient here ?  You would be constructing a lot of temporary strings otherwise.

Reply:
Yes, changed here 46b0e12